examples/Cloudy/Cloudy_example.jl

```diff
@@ -216,42 +253,40 @@ mcmc = MCMC.MCMCObj(yt_sample, Γy, priors, new_step, u0, max_iter,
                     mcmc_alg, burnin, svdflag=true)
 MCMC.sample_posterior!(mcmc, gpobj, max_iter)
 
-posterior = MCMC.get_posterior(mcmc)      
+posterior = MCMC.get_posterior(mcmc)
 
-post_mean = mean(posterior, dims=1)
-post_cov = cov(posterior, dims=1)
+post_mean = get_mean(posterior)
+post_cov = get_cov(posterior)
 println("post_mean")
 println(post_mean)
 println("post_cov")
 println(post_cov)
-println("D util")
-println(det(inv(post_cov)))
-println(" ")
 
 # Plot the posteriors together with the priors and the true parameter values
-true_values = [log(N0_true) log(θ_true) log(k_true)]
-n_params = length(true_values)
+n_params = length(names)
 
 for idx in 1:n_params
     if idx == 1
-        param = "N0"
-        xs = collect(4.5:0.01:6.5)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     elseif idx == 2
-        param = "Theta"
-        xs = collect(-1.0:0.01:2.5)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     elseif idx == 3
-        param = "k"
-        xs = collect(-4.0:0.01:1.0)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     else
         throw("not implemented")
     end
 
-    label = "true " * param
-    histogram(posterior[:, idx], bins=100, normed=true, fill=:slategray, 
+    label = "true " * names[idx]
+    println("size of samples to be plotted in histogram")
+    println(size(posterior.distributions[idx].distribution_samples))
+    ys = dropdims(posterior.distributions[idx].distribution_samples, dims=1)
+    histogram(ys, bins=100, normed=true, fill=:slategray, thickness_scaling=2.0,
               lab="posterior")
-    plot!(xs, mcmc.prior[idx].dist, w=2.6, color=:blue, lab="prior")
-    plot!([true_values[idx]], seriestype="vline", w=2.6, lab=label)
+    println("created histogram")
+    plot!(xs, get_distribution(mcmc.prior.distributions[idx]), w=2.6,
```

Comment:
Could we possibly have this as `get_distribution(mcmc.prior)` which provides a dictionary of the distributions - we then plot from the dictionary?

Reply:
Yep, done! I was thinking that for consistency it would be nice if `get_distribution()` also returned a dictionary (rather than the message "Contains samples only") for `Samples`. In the case of `Samples`, the dictionary's values could either be the `Samples` object itself or the samples in the form of an array. 
This would remove the inconsistency in how the prior and the posterior distribution are accessed here - the former is accessed using `get_distribution`, the latter via accessing the member itself (`posterior.distributions[idx].distribution_samples`)

Reply:
Implemented the change to `get_distribution` suggested above.